src/js/media-controller.js

```diff
@@ -278,6 +271,46 @@ class MediaController extends MediaContainer {
     propagateMediaState(this.associatedElements, stateName, state);
   }
 
+  associateDescendantsOf(element) {
```

Comment:
So the `associateElement` function was intended to be used for this. If we need two different functions for this purpose, I'd rather use `associateElement` for the one people will use and another name for the internal association function.

Could we just use the same function for both? If we don't use the subtree mutation observer then it seems all direct children could be associated in the same way as external elements.

Reply:
ok gonna chew on that. `associateElement` in the current version of the architecture was intentionally less presumptuous. I think there are benefits to having some method that is just for 1) "I want this very specific element associated and nothing else" vs. 2) "I want to monitor this and all descendants of this for potentially associating/unassociating". I'll rename/move bits around so `associateElement` is version (2) and rename (1) to something else unless you feel strongly otherwise. I'll also add this functionality to all of our Media UI Elements per your other comment.

Reply:
Cool, thank you. I'm not convinced there's a use case for "this element only and nothing inside it". Get me there and we'll talk about adding it. In the mean time I'm fine keeping the paths separate internally. I'm not excited about people having to learn and choose between two paths until we know we need that.